main.tf

```diff
@@ -61,8 +76,9 @@ resource "time_sleep" "wait_for_scc_cos_authorization_policy" {
 
 # attach a COS bucket and an event notifications instance
 resource "ibm_scc_instance_settings" "scc_instance_settings" {
+  count       = var.existing_scc_instance_crn == null ? 1 : 0
```

Comment:
this means that a user will be unable to an event notifications CRN to an existing SCC instance. I think we will want to support that use case, so you should probably use some dynamic blocks here to allow that to work

Reply:
Tried with dynamic blocks as follows:

```
  dynamic "event_notifications" {
    for_each = var.en_instance_crn != null ? { crn = var.en_instance_crn } : {}
    content {
      instance_crn = var.en_instance_crn
    }
  }
```

But this does not seem to work, as both the `event_notifications` and `object_storage` blocks are required:

```
 │ Error: Insufficient event_notifications blocks
│ 
│   on ../../main.tf line 83, in resource "ibm_scc_instance_settings" "scc_instance_settings":
│   83: resource "ibm_scc_instance_settings" "scc_instance_settings" {
│ 
│ At least 1 "event_notifications" block is required.
```

Instead, used a conditional operator which is working fine:

```
  event_notifications {
    instance_crn = var.en_instance_crn != null ? var.en_instance_crn : null
  }
  object_storage {
    instance_crn = var.cos_instance_crn != null ? var.cos_instance_crn : null
    bucket       = var.cos_bucket != null ? var.cos_bucket : null
  }
```

I also had to make `var.cos_instance_crn` and `var.cos_bucket` optional with the validation that either both have to be null or both have to be not null.

Reply:
@ocofaigh Okay, this will actually override the existing setting. I guess it might be a bad idea. I think we can add a new variable here.

Reply:
@ocofaigh - there is a problem with using existing Event Notification instance with SCC.
It turns out that when the ibm_scc_instance_settings resource is used, it creates a source in the EN instance with the same name "compliance", so if two SCC instances are provisioned with this module, the second one trying to use EN will fails with the error about "source with the same name already exists. 

It probably has to be fixed at the provider level as there is no name parameter in [SCC instance settings resource](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/scc_instance_settings#argument-reference), but when the integration is created in SCC UI, the source name is set to SCC instance name.

Reply:
@ocofaigh @in-1911 I have removed the logic to set `scc_instance_settings` when using an existing scc instance for now, we could add it later once we have fix in provider